include/circt/Dialect/FIRRTL/FIRRTLAnnotationHelper.h

```diff
@@ -84,6 +84,33 @@ struct AnnoTargetCache {
     return targets.lookup(name);
   }
 
+  /// Replace `oldOp` with `newOp` in the target cache. The new and old ops can
+  /// have different names.
+  void replaceOp(Operation *oldOp, Operation *newOp) {
+    auto namOp = dyn_cast<FNamableOp>(oldOp);
+    if (!namOp)
+      return;
+    if (auto name = namOp.getNameAttr(); name && !name.getValue().empty())
+      targets.erase(name);
+    if (auto name = cast<FNamableOp>(newOp).getNameAttr();
+        name && !name.getValue().empty())
+      targets.insert({name, OpAnnoTarget(newOp)});
+  }
+
+  /// Add a new module port to the target cache.
+  void insertPorts(FModuleLike mod, size_t portNo) {
```

Comment:
```suggestion
  void insertPort(FModuleLike mod, size_t portNo) {
```
Only inserts a single port.